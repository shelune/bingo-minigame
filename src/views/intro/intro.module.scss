@import "../../styles/utils/index";

.content {
  min-height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
}

.dialog {
  min-height: 500px;
  max-width: 550px;
  margin: 0 auto;
}

.formControl {
  width: 100%;
  margin-bottom: 20px;
  position: relative;
}

.label {
  @include text-label;
  font-weight: 600;
  color: $black;
  margin-bottom: 8px;
  display: inline-block;
}

.textInput {
  @include rounded-4;
  @include text-body;
  width: 100%;
  max-width: 100%;
  padding: 8px;
  border: 1px solid $gray;

  &:focus {
    outline-color: $orange;
  }
}

.formFunctions {
  text-align: center;
}

.button {
  @include text-body;
  @include rounded-4;
  cursor: pointer;
  background-color: $orange;
  color: $white;
  outline: none;
  border: none;
  height: 48px;
  line-height: 48px;
  min-width: 120px;

  &:disabled {
    cursor: not-allowed;
    background-color: $gray;
  }
}
