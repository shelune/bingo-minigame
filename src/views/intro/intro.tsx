import React, { useState } from "react";
import classNames from "classnames";

import css from "./intro.module.scss";

type Props = {
  setUsername: (name: string) => void;
  setShowingGame: (show: boolean) => void;
  username: string;
};

export default function IntroView({
  setUsername,
  setShowingGame,
  username,
}: Props) {
  const [error, setError] = useState("");

  return (
    <div className={css.content}>
      <div className={css.dialog}>
        <h1>Let's play Bingo!</h1>
        <form
          className={css.form}
          onSubmit={(e) => {
            e.preventDefault();
            // eslint-disable-next-line @typescript-eslint/no-floating-promises
            setShowingGame(true);
          }}
        >
          <div className={css.formControl}>
            <label htmlFor="username" className={css.label}>
              Player name
            </label>
            <input
              className={classNames(css.textInput, {
                [css.isEditing]: !!username,
              })}
              type="text"
              value={username}
              id="username"
              name="username"
              onChange={(e) => {
                setUsername(e.target.value);
              }}
            />
          </div>
          {error && (
            <div data-testid="login-error" className={css.formError}>
              {error}
            </div>
          )}
          <div className={css.formFunctions}>
            <button
              type="submit"
              className={css.button}
              data-testid="login-submit"
              disabled={!username}
            >
              Enter
            </button>
          </div>
        </form>
      </div>
    </div>
  );
}
