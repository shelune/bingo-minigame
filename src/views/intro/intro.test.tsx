import React from "react";
import { screen, fireEvent, render } from "@testing-library/react";

import { IntroView } from "./intro";

describe("IntroView", () => {
  it("should render", () => {
    const { container } = render(
      <IntroView
        setUsername={jest.fn()}
        setShowingGame={jest.fn()}
        username="Stranger"
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("should disable the button if no username", () => {
    render(
      <IntroView
        setUsername={jest.fn()}
        setShowingGame={jest.fn()}
        username=""
      />
    );
    expect(screen.getByText("Enter")).toHaveAttribute("disabled");
  });

  it("should call setUsername when typing in username", () => {
    const mockSetUsername = jest.fn();
    render(
      <IntroView
        setUsername={mockSetUsername}
        setShowingGame={jest.fn()}
        username=""
      />
    );
    fireEvent.change(screen.getByLabelText("Player name"), {
      target: {
        value: "Foobar",
      },
    });
    expect(mockSetUsername).toHaveBeenCalled();
  });

  it("should call setShowingGame when hitting Enter", () => {
    const mockSetShowingGame = jest.fn();
    render(
      <IntroView
        setUsername={jest.fn()}
        setShowingGame={mockSetShowingGame}
        username="Foobar"
      />
    );
    screen.getByText("Enter").click();
    expect(mockSetShowingGame).toHaveBeenCalled();
  });
});
