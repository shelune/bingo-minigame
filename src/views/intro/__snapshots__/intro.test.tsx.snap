// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IntroView should render 1`] = `
<div>
  <div
    class="content"
  >
    <div
      class="dialog"
    >
      <h1>
        Let's play Bingo!
      </h1>
      <form
        class="form"
      >
        <div
          class="formControl"
        >
          <label
            class="label"
            for="username"
          >
            Player name
          </label>
          <input
            class="textInput isEditing"
            id="username"
            name="username"
            type="text"
            value="Stranger"
          />
        </div>
        <div
          class="formFunctions"
        >
          <button
            class="button"
            data-testid="login-submit"
            type="submit"
          >
            Enter
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
